.github/workflows/main.yml

```diff
@@ -55,9 +58,40 @@ jobs:
           source "$CONDA/etc/profile.d/conda.sh"
           conda create -n psi4env python=${{ matrix.python-version }}
         shell: bash
-      - uses: ./.github/actions/install-main-dependencies
+      - name: Get Terra main last commit id
+        run: |
+          echo "TERRA_HASH=$(git ls-remote --heads https://github.com/Qiskit/qiskit-terra.git refs/heads/main  | awk '{print $1}')" >> $GITHUB_ENV
+        if: ${{ !startsWith(github.ref, 'refs/heads/stable') && !startsWith(github.base_ref, 'stable/') }}
+        shell: bash
+      - name: Terra Cache
+        id: terra-cache
+        uses: actions/cache@v3
+        with:
+          path: terra-cache
+          key: terra-cache-${{ matrix.os }}-${{ matrix.python-version }}-${{ env.TERRA_HASH }}-${{ env.CACHE_VERSION }}
+        if: ${{ !startsWith(github.ref, 'refs/heads/stable') && !startsWith(github.base_ref, 'stable/') }}
+      - uses: ./.github/actions/install-main-terra
+        with:
+          cache-path: terra-cache
+          cache-hit: ${{ steps.terra-cache.outputs.cache-hit }}
+        if: ${{ !startsWith(github.ref, 'refs/heads/stable') && !startsWith(github.base_ref, 'stable/') }}
+      - name: Get Aer main last commit id
+        run: |
+          echo "AER_HASH=$(git ls-remote --heads https://github.com/Qiskit/qiskit-aer.git refs/heads/main  | awk '{print $1}')" >> $GITHUB_ENV
+        if: ${{ !startsWith(github.ref, 'refs/heads/stable') && !startsWith(github.base_ref, 'stable/') }}
+        shell: bash
+      - name: Aer Cache
+        id: aer-cache
+        uses: actions/cache@v3
+        with:
+          path: aer-cache
+          key: aer-cache-${{ matrix.os }}-${{ matrix.python-version }}-${{ env.AER_HASH }}-${{ env.CACHE_VERSION }}
+        if: ${{ !startsWith(github.ref, 'refs/heads/stable') && !startsWith(github.base_ref, 'stable/') }}
+      - uses: ./.github/actions/install-main-aer
         with:
           os: ${{ matrix.os }}
+          cache-path: aer-cache
+          cache-hit: ${{ steps.aer-cache.outputs.cache-hit }}
```

Comment:
Is it not possible to put all of this logic (which is now duplicated 3 times) into a common place like e.g. the old `install-main-dependencies/action.yml`?

Reply:
The cache action needs to trigger a post action step in the main workflow in order to save the cache. When I tried initially to have inside the action, it didn't trigger it but I will try again since I changed other things after the last time I tested.